server/asset/common.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2019, The Decred developers
+// See LICENSE for details.
+
+package asset
+
+type Error string
+
+func (err Error) Error() string { return string(err) }
+
+const (
+	UnsupportedScriptError = Error("unsupported script type")
+)
+
+// The DEXAsset interface is an interface for a blockchain backend. The DEX
+// primarily needs to track UTXOs and transactions to validate orders, and
+// to monitor community conduct.
+type DEXAsset interface {
+	// UTXO should return a UTXO only for outputs that would be spendable on the
+	// blockchain immediately.
+	UTXO(txid string, vout uint32) (UTXO, error)
+	// BlockChannel creates and returns a new channel on which to receive updates
+	// when new blocks are connected.
+	BlockChannel(size int) chan uint32
```

Comment:
Not sure if people like this pattern or not. The DEX will definitely want to know when a block is received.

Reply:
Generally fine with me.  Depends how the pieces fit together though.  We can change it if needed.